### PR TITLE
chore: add tests for JsRequest

### LINKED ISSUE #1410: chore: add tests for JsRequest
The following tests are needed for `JsRequest`

**Unit Tests**
- Add unit tests for `impl From<&reqwest::Url> for Uri` with 100% coverage
- Add unit tests for `impl Display for Uri` with 100% coverage

**Integration Tests**
Write these in `.md` format in `tests/execution`.
- Make sure the JSRequest is correctly sent to JS.
- Make sure the JSRequest is correctly sent back from JS to Rust.

### PR DESCRIPTION
**Pull request by lazyfuhrer**
**Summary:**  
This PR adds the following unit and integration tests for JsRequest

**Issue Reference(s):**  
Fixes #1410
/claim #1410 

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: tests/execution/js_request-to-js-runtime.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# js_request-to-js-runtime
+
+```graphql @server
+schema @server(headers: {experimental: ["x-tailcall"]}) @upstream(baseURL: "http://example.com") {
```
Comment (tusharmath) [inline tests/execution/js_request-to-js-runtime.md:4]: This is not using any JS code.

Comment (lazyfuhrer) [reply tests/execution/js_request-to-js-runtime.md:4]: ig I've to use the in-build node env using ```@link``` ?

**Comment by tusharmath** [comment]
@lazyfuhrer Are u on this? I don't see any real activity in this PR.

**Comment by lazyfuhrer** [comment]
you can close this one. got busy with other work

**Pull request closed without merging**